source/features/vertical-front-matter.tsx

```diff
@@ -9,15 +9,24 @@ import features from '.';
 const hasFrontMatter = (): boolean => pageDetect.isSingleFile() && /\.(mdx?|mkdn?|mdwn|mdown|markdown|litcoffee)$/.test(location.pathname);
 
 function init(): void | false {
-	const table = select('[data-table-type="yaml-metadata"]')!;
+	const table = select('.markdown-body > table:first-child');
+	if (!table) {
+		return false;
+	}
+
 	const headers = select.all(':scope > thead th', table);
 	if (headers.length <= 4) {
 		return false;
 	}
 
-	const values = select.all(':scope > tbody > tr > td', table);
+	const rows = select.all(':scope > tbody > tr', table);
+	if (rows.length !== 1 || headers.length !== rows[0].childElementCount) {
```

Comment:
I think I missed one point: all rows always have the same number of cells, so the second check is necessary.

@yakov116 Could you add this into #4276?

```suggestion
	if (rows.length !== 1) {
```

Reply:
Yes

Reply:
I was going to say this, but technically tables could have spanning headers:


<table>
<tr>
	<th colspan="3"> title
<tr>
	<td>1
	<td>2
	<td>3
</table>

I thought it was an extra check to ensure we got exact tables

Reply:
🤦You are right, I forgot there's `colspan`. So the check is indeed necessary (not possible through Markdown though).

@yakov116 sorry for the trouble but we better revert the change 😅